packaging/build-package.sh

```diff
@@ -0,0 +1,150 @@
+#!/bin/bash
+
+set -euo pipefail
+
+function help () {
+  cat <<DOC
+Usage: TARGET=<target> VERSION=<version> $0
+
+Builds package for a given target and puts it into ./packaging/out.
+
+ENVIRONMENT VARIABLES
+
+  TARGET (required)
+    The packaging target. Available targets are
+
+    - debian
+    - pacman
+
+  VERSION (required)
+    Version of the package.
+
+  STACK_USE_DOCKER
+    If set to '1' then 'stack' is invoked with the '--docker' flag.
+    This means the build outputs from 'stack build --docker' are
+    used.
+DOC
+}
+
+
+# Wrapper for the stack that uses docker if `STACK_USE_DOCKER` is set
+# to `1`.
+function stack () {
+  if [ "${STACK_USE_DOCKER:-0}" = "1" ]; then
+    /usr/bin/env stack --docker "$@"
+  else
+    /usr/bin/env stack "$@"
+  fi
+}
+
+project_dir=$(realpath "$(dirname $BASH_SOURCE)/..")
+radicle_bindir=/usr/lib/radicle/bin
+radpath=/usr/lib/radicle/modules
+stack_bin_install_dir=$(stack path --local-install-root)/bin
+
+# Creates an executable file `$1` with content from stdin.
+function install-stdin () {
+  local target=$1
+  mkdir -p $(dirname "$target")
+  cat >"$target"
+  chmod +x "$target"
+}
+
+# Download the `ipfs` binary and put it into the directory `$1`.
+function get-ipfs () {
+  local target_dir=$1
+  mkdir -p "$target_dir"
+  curl \
+    --fail \
+    --silent \
+    --show-error \
+    --location \
+    "https://dist.ipfs.io/go-ipfs/v0.4.18/go-ipfs_v0.4.18_linux-amd64.tar.gz" \
+  | tar -xz --strip-components=1 -C "$target_dir" go-ipfs/ipfs
+}
+
+function prepare-package-root () {
+  install-stdin "$package_root/usr/bin/rad" <<EOF
+#!/bin/sh
+export radicle_bindir=$radicle_bindir
+export RADPATH=\${RADPATH:-$radpath}
+exec $radicle_bindir/rad "\$@"
+EOF
+
+  install-stdin "$package_root/usr/bin/radicle" <<EOF
+#!/bin/sh
+export radicle_bindir=$radicle_bindir
+export RADPATH=\${RADPATH:-$radpath}
+exec $radicle_bindir/radicle "\$@"
+EOF
+
+  install -D $(stack exec -- which git-remote-ipfs) -t "$package_root/usr/bin"
```

Comment:
How does this work? Is `stack` building the `git-remote-ipfs` executable implicitly?

Reply:
`git-remote-ipfs` is a dependency for the tests and thus installed

https://github.com/oscoin/radicle/blob/b979ca6237185ee3b38366f63ede7c4b06c4bc88/package.yaml#L116-L117